tests/linearizability/failpoints.go

```diff
@@ -239,11 +281,31 @@ type randomFailpoint struct {
 }
 
 func (f randomFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
-	failpoint := f.failpoints[rand.Int()%len(f.failpoints)]
+	gofailpointsAvailable, err := listFailpoints(clus.Procs[0])
+	if err != nil {
+		return err
+	}
+	availableFailpoints := make([]Failpoint, 0, len(f.failpoints))
+	for _, failpoint := range f.failpoints {
+		var available bool
+		if gofailpoint, ok := failpoint.(goPanicFailpoint); ok {
+			available = gofailpoint.available(gofailpointsAvailable)
+		} else {
+			available = failpoint.Available(clus.Procs[0])
```

Comment:
get all the available failpoints (issue a http request) again in this case?

Reply:
Don't understand the question.

Reply:
You already get all the available failpoints at line 284, and `(goPanicFailpoint) Available` gets all the failpoints again (line 175). 

Do not get the point why do you process `goPanicFailpoint` separately. Please add comment to explain the reason if you have a good one.

Reply:
Because each call to function `Available` makes http call to get list of failpoints. Instead of calling it for each go failpoint, I just get available failpoints once and call internal function `available` that takes a list of strings.

Reply:
I still do not get the answer to my original question. Why do you get the list of failpoints again when it isn't `goPanicFailpoint` (line 294)?

Reply:
I think the question is syndrome of a wider issue: 
   - gofailpoints are subset of all 'failpoints'. There are not go failpoints (like network latency failpoints). 
   - the introduced mechanism takes care only of gofailpoints. 

1.  A binary is awere of all gofailpoints (panic / sleep) it exposes - i.e. lines where it can be triggered. 
2. A binary might expose some fail-points not related to specific line of code (e.g. FS access problems assuming file interface is proxied within the binary itself). 
3. There are failpoints provided externally (e.g. by proxy) that the binary is not aware of. 

Abstractly thinking:
  1.  Binary can expose `failpointManifests` that describes all failpoints it supports (internally). 
  2. ALL the endpoints should be able to take the failpoints manifests (some will ignore it) and say whether they are supported on the particular binary. 
  3. Failpoint manifest could get cached within https://pkg.go.dev/go.etcd.io/etcd/tests/v3/framework/e2e#EtcdProcess. 
such that public `Failpoint.Available(etcdProcess)` method could has as simple API as it currently has.